python/lsst/fgcmcal/fgcmBuildStarsBase.py

```diff
@@ -0,0 +1,810 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""Base class for BuildStars using src tables or sourceTable_visit tables.
+"""
+
+import os
+import sys
+import traceback
+
+import numpy as np
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import lsst.afw.table as afwTable
+import lsst.geom as geom
+from lsst.daf.base import PropertyList
+from lsst.daf.base.dateTime import DateTime
+from lsst.meas.algorithms.sourceSelector import sourceSelectorRegistry
+
+from .utilities import computeApertureRadius
+from .fgcmLoadReferenceCatalog import FgcmLoadReferenceCatalogTask
+
+import fgcm
+
+REFSTARS_FORMAT_VERSION = 1
+
+__all__ = ['FgcmBuildStarsConfigBase', 'FgcmBuildStarsRunner', 'FgcmBuildStarsBaseTask']
+
+
+class FgcmBuildStarsConfigBase(pexConfig.Config):
+    """Base config for FgcmBuildStars tasks"""
+
+    instFluxField = pexConfig.Field(
+        doc=("Name of the source instFlux field to use.  The associated flag field "
+             "('<name>_flag') will be implicitly included in badFlags"),
+        dtype=str,
+        default='slot_CalibFlux_instFlux',
+    )
+    minPerBand = pexConfig.Field(
+        doc="Minimum observations per band",
+        dtype=int,
+        default=2,
+    )
+    matchRadius = pexConfig.Field(
+        doc="Match radius (arcseconds)",
+        dtype=float,
+        default=1.0,
+    )
+    isolationRadius = pexConfig.Field(
+        doc="Isolation radius (arcseconds)",
+        dtype=float,
+        default=2.0,
+    )
+    densityCutNside = pexConfig.Field(
+        doc="Density cut healpix nside",
+        dtype=int,
+        default=128,
+    )
+    densityCutMaxPerPixel = pexConfig.Field(
+        doc="Density cut number of stars per pixel",
+        dtype=int,
+        default=1000,
+    )
+    matchNside = pexConfig.Field(
+        doc="Healpix Nside for matching",
+        dtype=int,
+        default=4096,
+    )
+    coarseNside = pexConfig.Field(
+        doc="Healpix coarse Nside for partitioning matches",
+        dtype=int,
+        default=8,
+    )
+    filterMap = pexConfig.DictField(
+        doc="Mapping from 'filterName' to band.",
+        keytype=str,
+        itemtype=str,
+        default={},
+    )
+    requiredBands = pexConfig.ListField(
+        doc="Bands required for each star",
+        dtype=str,
+        default=(),
+    )
+    primaryBands = pexConfig.ListField(
+        doc=("Bands for 'primary' star matches. "
+             "A star must be observed in one of these bands to be considered "
+             "as a calibration star."),
+        dtype=str,
+        default=None
+    )
+    visitDataRefName = pexConfig.Field(
+        doc="dataRef name for the 'visit' field",
```

Comment:
This is because I was told years ago that I couldn't rely on "visit" being the name of "visit" forever, and this should be configurable.  I've found out later that "visit" is so hard-coded in certain parts of code that it can't be changed (as the expId experience showed).  But at least `fgcmcal` isn't going to be broken if the CCB decides to change the name!